api/pkg/apis/v1alpha1/utils/symphony-api.go

```diff
@@ -67,6 +68,70 @@ func GetSymphonyAPIAddressBase() string {
 	return symphonyAPIAddressBase
 }
 
+var getApiClientOnce sync.Once
+var apiClient *APIClient
+var err error
+
+func GetApiClient() (*APIClient, error) {
+	getApiClientOnce.Do(func() {
+		apiClient, err = getApiClient()
+	})
+	return apiClient, err
+}
+
+// For testing purpose only
+func UpdateApiClientUrl(url string) {
+	if apiClient == nil {
+		GetApiClient()
+	}
+	apiClient.baseUrl = url
+}
+
+func getApiClient() (*APIClient, error) {
+	clientOptions := make([]ApiClientOption, 0)
+	baseUrl := GetSymphonyAPIAddressBase()
+	if err != nil {
+		return nil, err
+	}
+	if caCert, ok := os.LookupEnv(constants.ApiCertEnvName); ok {
+		clientOptions = append(clientOptions, WithCertAuth(caCert))
+	}
+
+	if ShouldUseSATokens() {
+		clientOptions = append(clientOptions, WithServiceAccountToken())
+	} else {
+		user := "admin"
```

Comment:
These should come from configuration file instead of hard coded here. For standalone mode, we still want user to be able to leverage the built-in user store.

Reply:
I got your point. I'll keep the singleton and change the tokenprovider function to allow username and pwd configurable. Working on the fix now.